expression/constant.go

```diff
@@ -50,15 +50,31 @@ var (
 
 // Constant stands for a constant value.
 type Constant struct {
-	Value        types.Datum
-	RetType      *types.FieldType
-	DeferredExpr Expression // parameter getter expression
-	hashcode     []byte
+	Value         types.Datum
+	RetType       *types.FieldType
+	DeferredExpr  Expression // parameter getter expression
+	DeferredParam *DeferredParam
```

Comment:
Could you add more comments about the `DeferredExpr` and `DeferredParam`? Specially on how they are worked together.